wgengine/netstack/netstack.go

```diff
@@ -913,13 +921,19 @@ func (ns *Impl) acceptTCP(r *tcp.ForwarderRequest) {
 		}
 	}
 
-	if ns.ForwardTCPIn != nil {
-		c := createConn()
-		if c == nil {
-			return
+	if ns.GetTCPHandlerForFlow != nil {
+		handler, ok := ns.GetTCPHandlerForFlow(clientRemoteAddrPort, dstAddrPort)
+		if ok {
+			if handler == nil {
+				r.Complete(true)
+				return
+			}
+			c := createConn()
+			if c == nil {
+				return
```

Comment:
will createConn on failure r.Complete(true) to send the RST too?

Reply:
(I wrote this) Yep, it should:
https://github.com/tailscale/tailscale/blob/c09578d060e001453c43c78d492891ad92bc862e/wgengine/netstack/netstack.go#L817-L823

Reply:
yeah, added a comment at the call site